### PR TITLE
Peg/pay 1549 accept collectionidclientid interchangably in sdk

### PR DESCRIPTION
**Pull request by pgarciaegido**
https://linear.app/crossmint/issue/PAY-1549/accept-collectionidclientid-interchangably-in-sdk

Making collectionId optional as a prop to all our SDKs. 

Before merging:

- [x] Manually test Vanilla SDK
- [x] Next JS starter
- [x] React starter
- [x] React Payment Element
- [x] Vue starter

Screenshot 2023-06-22 at 12 45 05
Screenshot 2023-06-22 at 12 51 51

### CODE REVIEW

File: packages/ui/vue-ui/src/components/CrossmintPaymentElement.vue [hunk 1]
```diff
@@ -17,7 +17,8 @@ import { crossmintPaymentService, crossmintUiService } from "@crossmint/client-s
 // TODO: Looks like you cannot import the interface directly from the package
 // https://github.com/vuejs/core/issues/4294#issuecomment-970861525
 export interface PaymentElement {
-    clientId: string;
+    clientId?: string;
```
Comment (mPaella) [inline packages/ui/vue-ui/src/components/CrossmintPaymentElement.vue:20]: union type here

Comment (pgarciaegido) [reply packages/ui/vue-ui/src/components/CrossmintPaymentElement.vue:20]: not working... according to chatGPT is a limitation on Vue 3 defineProps

Comment (pgarciaegido) [reply packages/ui/vue-ui/src/components/CrossmintPaymentElement.vue:20]: Screenshot 2023-06-21 at 20 25 18

Comment (pgarciaegido) [reply packages/ui/vue-ui/src/components/CrossmintPaymentElement.vue:20]: I think Vue 3.3 comes with a lot of improvements when it comes to typescript... we might need to try to upgrade and see what happens. In any case, I will add a manual check in the vue package.

File: packages/ui/react-ui/src/CrossmintPayButton.tsx [hunk 2]
```diff
@@ -19,41 +19,44 @@ const defaultMintConfig: any = {
...
+    const collectionId = "clientId" in props ? props.clientId : props.collectionId;
+
     const [connecting, setConnecting] = useState(false);
     const [status, setStatus] = useState(onboardingRequestStatusResponse.WAITING_SUBMISSION);
     const { isServerSideRendering } = useEnvironment();
 
     const { fetchClientIntegration } = crossmintStatusService({
         libVersion: LIB_VERSION,
-        clientId,
+        clientId: collectionId!,
```
Comment (mPaella) [inline packages/ui/react-ui/src/CrossmintPayButton.tsx:59]: do we really need `!`

**Review by mPaella — APPROVED** [review]

**Pull request merged**